libs/entitlement-runtime/src/test/java/org/elasticsearch/entitlement/policy/PolicyBuilderTests.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.policy;
+
+import org.elasticsearch.test.ESTestCase;
+
+public class PolicyBuilderTests extends ESTestCase {
+
+    public void testPolicyBuilder() {
+        new PolicyBuilder("test-policy.yaml", PolicyBuilderTests.class.getResourceAsStream("test-policy.yaml")).buildPolicy();
```

Comment:
This needs some sad-path tests too, with various kinds of invalid YAML structures.

Reply:
Definitely!

Reply:
In the past I have had some success with a tool called pitest to inspire unit test cases based on the sorts of bugs they would detect. It can be tricky to set up though.